src/TLuaInterpreter.cpp

```diff
@@ -12721,29 +12752,40 @@ int TLuaInterpreter::putHTTP(lua_State* L)
 
     QByteArray fileToUpload;
     QString fileLocation;
-    if (!lua_isstring(L, 4) && !lua_isnoneornil(L, 4)) {
-        lua_pushfstring(L, "putHTTP: bad argument #4 type (file to send as string location expected, got %s!)", luaL_typename(L, 4));
+    if (!lua_isstring(L, pos + 4) && !lua_isnoneornil(L, pos + 4)) {
+        lua_pushfstring(L, "%s: bad argument #%d type (file to send as string location expected, got %s!)", functionName, pos + 4, luaL_typename(L, 4));
         return lua_error(L);
     }
-    if (lua_isstring(L, 4)) {
-        fileLocation = lua_tostring(L, 4);
+    if (lua_isstring(L, pos + 4)) {
+        fileLocation = lua_tostring(L, pos + 4);
     }
 
     if (!fileLocation.isEmpty()) {
         QFile file(fileLocation);
         if (!file.open(QFile::ReadOnly)) {
-            return warnArgumentValue(L, __func__, QStringLiteral("couldn't open '%1', is the location correct and do you have permissions to it?").arg(fileLocation));
+            return warnArgumentValue(L, functionName, QStringLiteral("couldn't open '%1', is the location correct and do you have permissions to it?").arg(fileLocation));
         }
 
         fileToUpload = file.readAll();
         file.close();
     }
 
     host.updateProxySettings(host.mLuaInterpreter.mpFileDownloader);
-    QNetworkReply* reply = host.mLuaInterpreter.mpFileDownloader->put(request, fileToUpload.isEmpty() ?dataToPost.toUtf8() : fileToUpload);
+
+    QNetworkReply* reply;
+    switch (operation) {
+        case QNetworkAccessManager::PostOperation:
+            reply = host.mLuaInterpreter.mpFileDownloader->post(request, fileToUpload.isEmpty() ?dataToPost.toUtf8() : fileToUpload);
+            break;
+        case QNetworkAccessManager::PutOperation:
+            reply = host.mLuaInterpreter.mpFileDownloader->put(request, fileToUpload.isEmpty() ?dataToPost.toUtf8() : fileToUpload);
+            break;
+        default:
+            reply = host.mLuaInterpreter.mpFileDownloader->sendCustomRequest(request, verb.toUtf8(), fileToUpload.isEmpty() ?dataToPost.toUtf8() : fileToUpload);
+    };
 
     if (mudlet::debugMode) {
-        TDebug(QColor(Qt::white), QColor(Qt::blue)) << "putHTTP: script is uploading data to " << reply->url().toString() << "\n" >> 0;
+        TDebug(QColor(Qt::white), QColor(Qt::blue)) << functionName << ": script is uploading data to " << reply->url().toString() << "\n" >> 0;
```

Comment:
Be advised that this will need revision when #4781 goes in - you will need to change the `0` at the end to a pointer to the `Host` for this profile - given that this is a `static` method you will need to use `&host` as you cannot access non-static members (such as `(QPointer<Host>) TLuaInterpreter::mpHost` from here.

Reply:
Since #4781 is still in review I am merging this one, not sure if you want to rebase or we can correct that one after merging it.